src/fermo/app_utils/input_testing.py

```diff
@@ -0,0 +1,287 @@
+from werkzeug.utils import secure_filename
+import os
+import json
+
+# for type hinting
+from typing import Tuple
+from werkzeug.datastructures import FileStorage
+
+
+def check_file_format(
+    filename: str,
+    file_format: str,
+    allowed_extensions: list,
+) -> (str):
+    ''' Check filename for validity
+
+    Parameters
+    ----------
+    filename: `str`
+    file_format: `str`
+        File format of one specific input with the leading dot
+    allowed_extension: `list`
+        List of strings as specified in config file
+
+    Return
+    ------
+    feedback: `str`
+        String with feedback for the user to be flashed
+    '''
+    if '.' not in file_format:
+        raise ValueError(
+            'file_format should be specified with the leading dot'
+        )
+    elif file_format not in allowed_extensions:
+        raise ValueError(
+            '''file_format must be an allowed_extension as specified in
+            the config'''
+        )
+    elif filename == '':
+        feedback = 'No file was loaded. Please upload a session-file.'
+    elif not filename.endswith(file_format):
+        feedback = f"File must be a {file_format}-file!"
+    else:
+        feedback = ''
+    return feedback
+
+
+def save_file(
+        file: FileStorage,
+        file_format: str,
+        allowed_extensions: list,
+        upload_folder: str,
+) -> Tuple[str, str]:
+    '''Save the file if it is valid
+
+    Parameters
+    ----------
+    file: `werkzeug.datastructures.FileStorage`
+        An element of request.files
+    file_format: `str`
+        File format of one specific input with the leading dot
+    allowed_extensions: `list`
+        List of str, as specified in config file
+    upload_folder: `str`
+        Path to location where files should be stored
+
+    Return
+    ------
+    `str`
+        Feedback for the user to be flashed or the secure filename of the saved
+        file
+    `bool`
+        True if file-saving was successfull, False otherwise
+
+    Notes
+    -----
+    Checks if a file was provided and if it has the right format (judging
+    by the filename). If so, the file is saved in the 'upload_folder'
+    '''
+    filename = secure_filename(file.filename)
+    feedback = check_file_format(
+            filename,
+            file_format,
+            allowed_extensions,
+        )
+    if feedback:
+        return feedback, False
+    else:
+        try:
+            file.save(os.path.join(upload_folder, filename))
+            return filename, True
+        except FileNotFoundError as e:
```

Comment:
Saving into folder is only a temporary solution, right?

Reply:
yes, the upload folder as set in the configurations is supposed to be replaced by a database. I'm not yet sure how the connection to a database will be implemented therefore this temporary solution